src/UID2.Client/UID2Encryption.cs

```diff
@@ -435,5 +436,20 @@ private static IdentityScope DecodeIdentityScopeV3(byte value)
         {
             return (IdentityScope)((value >> 4) & 1);
         }
+
+        private static IdentityType? GetIdentityType(byte idType)
```

Comment:
I think this function could encapsulate the logic of reading the token bytes and extracting the identity type rather than the caller having to do it